tests/test_dimension.py

```diff
@@ -831,6 +831,29 @@ def test_implicit_dims(self):
 
         assert np.all(f.data == F)
 
+    def test_stepping_dim_in_condition_lowering(self):
+        """
+        Check that the compiler performs the following lowering::
+
+        if f[t, x] > 2 {            if f[time, x] > 2 {
+          ...                -->      ...
+        }                           }
+        """
+        grid = Grid(shape=(4, 4))
+        _, y = grid.dimensions
+
+        ths = 10
+        g = TimeFunction(name='g', grid=grid)
+
+        ci = ConditionalDimension(name='ci', parent=y, condition=Le(g, ths))
```

Comment:
ANy reason why `y` is the parent?

Reply:
Innermost loop so that you nest all dims. But this is not a good answer.
I ask myself? So if x was parent what shoud the value of y index on the condition be?
Ideally I need to write an MFE here or other example to have conditionals iterating over one dim while other is constant. i.e. explicitly:

`         ci = ConditionalDimension(name='ci', parent=x, condition=Le(g[x,5], ths)) `
THoughts?

Reply:
I thought a bit more,

Would we like to have somethingl ike that?

ci = ConditionalDimension(name='ci', parent=x, condition=(g[x, 2] > ths))

Reply:
Possibly yes.

Reply:
So conceptually:
```
nx, ny = 4 , 4

shape = (nx, ny)
grid = Grid(shape=shape)

x, y = grid.dimensions

g = Function(name='g', grid=grid)
h = Function(name='h', grid=grid, dimensions=(x,), shape=(shape[0],))

cond = Le(h, 5)
ci = ConditionalDimension(name='ci', parent=x, condition=cond)

eq0 = Eq(h, g[x, 1])
eq1 = Eq(g, g + 1, implicit_dims=ci)
op = Operator([eq0, eq1])
print(op.ccode)
op.apply()
```

produces:
```
    for (int x = x_m; x <= x_M; x += 1)
    {
      h[x + 1] = g[x + 1][2];
      if (h[x + 1] <= 5)
      {
        #pragma omp simd aligned(g:32)
        for (int y = y_m; y <= y_M; y += 1)
        {
          g[x + 1][y + 1] = g[x + 1][y + 1] + 1;
        }
      }
    }
```
Ultimate goal should be to have CD able to be explicitly defined.
Currently it's not. SHould that be a different PR?

Reply:
Ideally, we need a way to get rid of `h`.